### ISSUE TITLE
Compilation fails in FreeBSD-13.0

### ISSUE THREAD

**Issue by Huskers2020** [Amzo/FreeBSD-Tensorflow#3]
When I try to compile the latest version of this code I get the following error, can you advise if you ran into a similar issue or suggest a solution. My progress is blocked for now.

Looking forward for your inputs - 

SUBCOMMAND: # @llvm-project//llvm:TableGen [action 'Compiling llvm/lib/TableGen/SetTheory.cpp', configuration: b163e161473125cddda9de10f3258d1c38f8c097d6ae21efec33fc5768658391, execution platform: @local_execution_config_platform//:platform]
(cd /usr/home/prv/Downloads/FreeBSD-Tensorflow-master/work-py39/bazel_out/dae3e1592a05ec9ba37a8572c11e389c/execroot/org_tensorflow && \
  exec env - \
    KERAS_HOME=/usr/home/prv/Downloads/FreeBSD-Tensorflow-master/work-py39/.keras \
    PATH=/usr/home/prv/Downloads/FreeBSD-Tensorflow-master/work-py39/.bin:/sbin:/bin:/usr/sbin:/usr/bin:/home/prv/.local/bin:/usr/local/sbin:/usr/local/bin:/home/prv/bin:/compat/linux/usr/lib/ \
    PWD=/proc/self/cwd \
  /usr/local/llvm-devel/bin/clang-16 -U_FORTIFY_SOURCE '-D_FORTIFY_SOURCE=1' -fstack-protector -Wall -fno-omit-frame-pointer -g0 -O2 -DNDEBUG -ffunction-sections -fdata-sections '-std=c++17' -MD -MF bazel-out/freebsd-opt-exec-50AE0418/bin/external/llvm-project/llvm/_objs/TableGen/SetTheory.d '-frandom-seed=bazel-out/freebsd-opt-exec-50AE0418/bin/external/llvm-project/llvm/_objs/TableGen/SetTheory.o' '-DLLVM_ON_UNIX=1' '-DHAVE_BACKTRACE=1' '-DBACKTRACE_HEADER=<execinfo.h>' '-DLTDL_SHLIB_EXT=".so"' '-DLLVM_PLUGIN_EXT=".so"' '-DLLVM_ENABLE_THREADS=1' '-DHAVE_DEREGISTER_FRAME=1' '-DHAVE_LIBPTHREAD=1' '-DHAVE_PTHREAD_GETNAME_NP=1' '-DHAVE_PTHREAD_GETSPECIFIC=1' '-DHAVE_PTHREAD_H=1' '-DHAVE_PTHREAD_SETNAME_NP=1' '-DHAVE_REGISTER_FRAME=1' '-DHAVE_SETENV_R=1' '-DHAVE_STRERROR_R=1' '-DHAVE_SYSEXITS_H=1' '-DHAVE_UNISTD_H=1' '-DLLVM_NATIVE_ARCH="X86"' '-DLLVM_NATIVE_ASMPARSER=LLVMInitializeX86AsmParser' '-DLLVM_NATIVE_ASMPRINTER=LLVMInitializeX86AsmPrinter' '-DLLVM_NATIVE_DISASSEMBLER=LLVMInitializeX86Disassembler' '-DLLVM_NATIVE_TARGET=LLVMInitializeX86Target' '-DLLVM_NATIVE_TARGETINFO=LLVMInitializeX86TargetInfo' '-DLLVM_NATIVE_TARGETMC=LLVMInitializeX86TargetMC' '-DLLVM_NATIVE_TARGETMCA=LLVMInitializeX86TargetMCA' '-DLLVM_HOST_TRIPLE="x86_64-unknown-freebsd13.0"' '-DLLVM_DEFAULT_TARGET_TRIPLE="x86_64-unknown-freebsd13.0"' -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -DBLAKE3_NO_AVX2 -DBLAKE3_NO_AVX512 -DBLAKE3_NO_SSE2 -DBLAKE3_NO_SSE41 '-DBLAKE3_USE_NEON=0' -iquote external/llvm-project -iquote bazel-out/freebsd-opt-exec-50AE0418/bin/external/llvm-project -iquote external/llvm_terminfo -iquote bazel-out/freebsd-opt-exec-50AE0418/bin/external/llvm_terminfo -iquote external/llvm_zlib -iquote bazel-out/freebsd-opt-exec-50AE0418/bin/external/llvm_zlib -isystem external/llvm-project/llvm/include -isystem bazel-out/freebsd-opt-exec-50AE0418/bin/external/llvm-project/llvm/include -g0 -I/usr/local/include -g0 -no-canonical-prefixes -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -c external/llvm-project/llvm/lib/TableGen/SetTheory.cpp -o bazel-out/freebsd-opt-exec-50AE0418/bin/external/llvm-project/llvm/_objs/TableGen/SetTheory.o)
# Configuration: b163e161473125cddda9de10f3258d1c38f8c097d6ae21efec33fc5768658391
# Execution platform: @local_execution_config_platform//:platform
INFO: Elapsed time: 455.809s, Critical Path: 0.40s
INFO: 68 processes: 25 internal, 43 local.
FAILED: Build did NOT complete successfully
*** Error code 1

**Comment by Huskers2020** [comment]
Issue Resolved - Had to install llvm
pkg install devel/llvm-devel

**Comment by Huskers2020** [comment]
issue resolved - compiling progress past this point

**Comment by Amzo** [comment]
I'll add a quick note here. To make full use of the keras API you will need the latest Keras package. Which I ported here
https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=266311#c2

```
>>> import platform
>>> platform.system()
'FreeBSD'
>>> import tensorflow as tf
>>> mnist = tf.keras.datasets.mnist
>>> (x_train, y_train), (x_test, y_test) = mnist.load_data()
>>> model = tf.keras.models.Sequential([
...   tf.keras.layers.Flatten(input_shape=(28, 28)),
...   tf.keras.layers.Dense(128, activation='relu'),
...   tf.keras.layers.Dropout(0.2),
...   tf.keras.layers.Dense(10)
... ])
>>> predictions = model(x_train[:1]).numpy()
>>> tf.nn.softmax(predictions).numpy()
array([[0.0000000e+00, 1.0000000e+00, 2.4388763e-36, 0.0000000e+00,
        0.0000000e+00, 0.0000000e+00, 0.0000000e+00, 0.0000000e+00,
        0.0000000e+00, 0.0000000e+00]], dtype=float32)
>>> loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)
>>> model.compile(optimizer='adam',
...               loss=loss_fn,
...               metrics=['accuracy'])
>>> model.fit(x_train, y_train, epochs=5)
Epoch 1/5
1875/1875 [==============================] - 2s 1ms/step - loss: 2.4908 - accuracy: 0.7472 
Epoch 2/5
1875/1875 [==============================] - 2s 1ms/step - loss: 0.6050 - accuracy: 0.8380
Epoch 3/5
1875/1875 [==============================] - 2s 1ms/step - loss: 0.4909 - accuracy: 0.8690
Epoch 4/5
1875/1875 [==============================] - 2s 1ms/step - loss: 0.4346 - accuracy: 0.8852
Epoch 5/5
1875/1875 [==============================] - 2s 1ms/step - loss: 0.4069 - accuracy: 0.8943
<keras.callbacks.History object at 0xa83866b80>
>>> 
```

**Comment by Huskers2020** [comment]
My compilation continued and completed successfully  except for the one issue I had encountered related llvm. Kudos to you for making this happen!

Post installation I ran into protobuf pyrwrap-tensorflow and grpcio related versions, TF 2.9 needs ver < 3.20 as per pip message, when I tried to import TF initialization failed with 3.19.

On updating protobuf to 4.21 initialization of TF worked, I have not tested the TG post this step as it was too late (1:00am)
